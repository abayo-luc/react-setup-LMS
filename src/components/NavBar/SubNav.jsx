import React from 'react';
import { Link } from 'react-router-dom';

const SubNav = () => (
  <div id="sub-nav">
    <div className="nav-links">
      <ul className="links">
        {[
          'Economic',
          'Culture',
          'Tech',
          'Politics',
          'Health',
          'Science',
          'Art',
        ].map((text, index) => (
          <li key={`${index + 1}-${text.slice(1, 3)}`}>
            <Link to="/">{text}</Link>
          </li>
        ))}
      </ul>
    </div>
  </div>
);

export default SubNav;
