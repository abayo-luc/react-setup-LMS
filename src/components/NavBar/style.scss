@mixin links {
  li {
    display: inline;
    margin: 0px 10px;
    font-size: 16px;
    text-transform: capitalize;
    @content;
  }
}
.full-nav {
  #nav {
    background-color: #007bff;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    padding: 0px 15px;
    position: fixed;
    z-index: 2;
    width: 100%;
    .title {
      color: #fff;
      font-size: 18px;
      line-height: normal;
      text-align: center;
    }
    .nav-links {
      float: right;
      ul.links {
        list-style: none;
        @include links {
          color: #fff;
        }
        a {
          text-anchor: unset;
          color: inherit;
          text-decoration: none;
          cursor: pointer;
        }
      }
    }
  }
  #sub-nav {
    display: flex;
    flex-direction: row;
    justify-content: center;
    padding-top: 60px;
    .nav-links {
      ul.links {
        list-style: none;
        @include links {
          a {
            color: #000;
            text-anchor: none;
            text-decoration: none;
            &:hover,
            &.active {
              color: #007bff;
            }
          }
        }
      }
    }
  }
}
