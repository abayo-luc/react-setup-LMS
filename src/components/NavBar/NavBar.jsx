import React from 'react';
import './style.scss';
import SubNav from './SubNav';

const icon = require('../../assets/icons/lines.svg');

const NavBar = () => (
  <div>
    <div id="nav">
      <div className="menu">
        <img src={icon} alt="-" />
      </div>
      <div className="title">
        <h2>ReadIT</h2>
      </div>
      <div className="nav-links">
        <ul className="links">
          <li>Home</li>
          <li>About</li>
          <li>Team</li>
        </ul>
      </div>
    </div>
    <SubNav />
  </div>
);
export default NavBar;
