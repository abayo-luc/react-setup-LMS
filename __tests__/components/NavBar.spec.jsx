import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NavBar from '../../src/components/NavBar/NavBar';

const jSonWarper = toJson(shallow(<NavBar />));

describe('NavBar Component', () => {
  test('should match the snapshot', () => {
    expect(jSonWarper).toMatchSnapshot();
  });
});
