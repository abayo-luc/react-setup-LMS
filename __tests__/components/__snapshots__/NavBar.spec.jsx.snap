// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component should match the snapshot 1`] = `
<div>
  <div
    id="nav"
  >
    <div
      className="menu"
    >
      <img
        alt="-"
        src={Object {}}
      />
    </div>
    <div
      className="title"
    >
      <h2>
        ReadIT
      </h2>
    </div>
    <div
      className="nav-links"
    >
      <ul
        className="links"
      >
        <li>
          Home
        </li>
        <li>
          About
        </li>
        <li>
          Team
        </li>
      </ul>
    </div>
  </div>
  <SubNav />
</div>
`;
